### PR TITLE
[4.0] Modules can't be created

### PR DESCRIPTION
**Pull request by laoneo**
It's not possible to create a module in J4. An error is displayed:
_Save failed with the following error: Field 'content' doesn't have a default value_

I guess it is not required to have a content for the module, so we can remove it from the table structure.

@alikon can you help me out with the update script of postgres?

### CODE REVIEW

**Comment by wilsonge** [comment]
Looks good to me. Can you do some update files as well please - I want to start gearing up for a 3.9 alpha 1 where you can upgrade to 4.0

**Comment by laoneo** [comment]
There is an update file. Only the one for postgres is missing.

File: installation/sql/postgresql/joomla.sql [hunk 1]
```diff
@@ -1442,7 +1442,7 @@ CREATE TABLE IF NOT EXISTS "#__modules" (
   "asset_id" bigint DEFAULT 0 NOT NULL,
   "title" varchar(100) DEFAULT '' NOT NULL,
   "note" varchar(255) DEFAULT '' NOT NULL,
-  "content" text DEFAULT '' NOT NULL,
+  "content" text DEFAULT '' NULL,
```
Comment (Bakual) [inline installation/sql/postgresql/joomla.sql]: It doesn't make sense to have a default value specified if the field is allowed to be NULL, right?

**Comment by alikon** [comment]
add in administrator/components/com_admin/sql/updates/postgresql/4.0.0-2018-03-05.sql
`ALTER TABLE "#__modules" ALTER COLUMN "content" DROP NOT NULL;`

but shouldn't work without #19707

p.s
unable to open pr on your branch

**Comment by laoneo** [comment]
Thanks @alikon, added the script. Can you double check the changes in the joomla.sql of postgres?  Are they ok?

File: installation/sql/postgresql/joomla.sql [hunk 2]
```diff
@@ -1442,7 +1442,7 @@ CREATE TABLE IF NOT EXISTS "#__modules" (
   "asset_id" bigint DEFAULT 0 NOT NULL,
   "title" varchar(100) DEFAULT '' NOT NULL,
   "note" varchar(255) DEFAULT '' NOT NULL,
-  "content" text DEFAULT '' NOT NULL,
+  "content" text NULL,
```
Comment (alikon) [inline installation/sql/postgresql/joomla.sql]: should be 
`"content" text, `

**Comment by laoneo** [comment]
I guess we are here fine.

**Comment by alikon** [comment]
#19707 apart yes

**Comment by anuragteapot** [comment]
@laoneo its fine modules <code>save success </code> :)

**Comment by wilsonge** [comment]
You need to use Alter Column in the mysql change please otherwise the schema checker (https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Schema/ChangeItem/MysqlChangeItem.php) won't pick it up

**Comment by laoneo** [comment]
There is no `Alter Column` or do I miss something here?

**Comment by wilsonge** [comment]
OK I'm just being dumb. We also have support for change https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Schema/ChangeItem/MysqlChangeItem.php#L168 

But looks like currently it's only going to work for changing column types. Not for swapping things to not null. In fact the "kludge" for integer unsigned is going to trigger with `TEXT NULL` tho right? https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Schema/ChangeItem/MysqlChangeItem.php#L170-L176

**Comment by brianteeman** [comment]
Isnt this the same as #17752 ??

**Comment by laoneo** [comment]
This one here is a subset of it and solves one purpose to fix creating modules. #17752 does more and needs carefully be tested as there should basically be a test case for every change in the install scripts which got changed. #17752 is also not finished as the update scripts are missing.

**Comment by laoneo** [comment]
Where do we stay here? I think it would be not bad when we release the next alpha to be able to create modules.

**Comment by Quy** [comment]
> But looks like currently it's only going to work for changing column types. Not for swapping things to not null. In fact the "kludge" for integer unsigned is going to trigger with TEXT NULL tho right? 

After PR and database fix, the modules table stayed the same.
Tested with 5.5.5-10.1.29-MariaDB

**Pull request merged**